zeth-contracts/contracts/BaseMixer.sol

```diff
@@ -116,50 +131,52 @@ contract BaseMixer is MerkleTreeMiMC7, ERC223ReceivingContract {
     // and modifies the state of the mixer contract accordingly
     // (ie: Appends the commitments to the tree, appends the nullifiers to the list and so on)
     function check_mkroot_nullifiers_hsig_append_nullifiers_state(
-        uint[2][2] memory vk,
+        uint[4] memory vk,
         uint[] memory primary_inputs) internal {
```

Comment:
This is a bit more invasive than it first appears.  `primary_inputs` is passed to several functions, including to the verifier.  The verifier cannot use constants defined on other contracts, so the only way to make this work would be to re-specify the number of inputs, outputs and the calculation of the number of public inputs.  i.e. we'd re-specify this info in 3 places, which is a bit of a maintenance nightmare when anything changes.

I'd been considering turning the verifier into a library anyway, to avoid the cost of the call.  A library also connot share constants from some other contract, so in the end the changes I've made to do this are:
- verifier methods and types moved into the mixer
- mixer now holds the verification key
- inputs array size set statically to the pre-existing constant `BaseMixer.nbInputs`

With this change there are now no other contracts to be deployed for a Zeth instance, which also simplifies the client code.